src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -99,12 +120,28 @@ public override TestResult Execute(TestExecutionContext context)
         {
             try
             {
-                if (!Task.Run(() => context.CurrentResult = innerCommand.Execute(context)).Wait(_timeout))
+                var testExecution = RunTestOnSeparateThread(context);
+
+                Wait(testExecution, _timeout);
+
+                if (testExecution.IsCompleted)
                 {
```

Comment:
Since we're changing the implementation from `.Wait(_timeout)`, I would suggest this:

```cs
if (Task.WaitAny(new[] { testExecution }, _timeout) != -1)
{
```

GetAwaiter().GetResult() on Task.WhenAny is not better than .Wait because Task.WhenAny is guaranteed to complete with TaskStatus.RanToCompletion. (https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.whenany?view=netcore-3.1#System_Threading_Tasks_Task_WhenAny_System_Threading_Tasks_Task___)

Reply:
Thank you, I did not realize I could use `WaitAny` for this.